### DISCUSSION TITLE
Ekuiper rules sending notifications service

### DISCUSSION THREAD

**Discussion by antiadominguez** [lf-edge/ekuiper#2932] [Q&A]
I see on the documentation that you can configure ekuiper rules engine to send notifications with the notifications service, but I don't know how to do it and I didn't find anything on the documentation. I tryed with a rest sink but I didn't send the notification. How do I configure the sink to send a email notification with the service.

**Comment by ngjaying** [comment]
@antiadominguez If your notification service is an HTTP service, REST sink can trigger it. Currently, it did not support to send an email directly.

**Reply by antiadominguez** [reply]
But I want to connect it o edgex notification service with REST but I don't know how to configure it. I tried this but it doesn't send any notification. 
![Screenshot from 2024-06-17 13-42-39](https://github.com/lf-edge/ekuiper/assets/165683333/b3b39e6c-273a-477d-9498-fb2047408b25)

**Reply by ngjaying** [reply]
ah, here is a tutorial about how to trigger the command service https://ekuiper.org/docs/en/latest/edgex/edgex_rule_engine_command.html. Trigger the notification service should be similar.

**Reply by antiadominguez** [reply]
I have tried that and I have the following error on logs, I think it's a bad configuration but I don't know how to do it, thats why I made my question
sink_rest_0_0_last_exception": "Post "[http://localhost:59860/api/v3/notification\](http://localhost:59860/api/v3/notification%5C)": dial tcp [::1]:59860: connect: connection refused: rest sink fails to send out the data: method=POST path="[http://localhost:59860/api/v3/notification\](http://localhost:59860/api/v3/notification%5C)" request_body=""{"apiVersion":"v3","notification":{"category":"daily-notify","content":"content","description":"devicenotice","labels":["daily"],"sender":"ekuiper","severity":"CRITICAL","status":"NEW"}}","",
![Screenshot from 2024-06-18 09-11-53](https://github.com/lf-edge/ekuiper/assets/165683333/fdf5ac79-7e61-4cd2-bcba-4b713373f10d)

**Reply by ngjaying** [reply]
eKuiper and notification service is not in the same machine, thus localhost is not the correct host endpoint. Try to use the real host ip.

**Reply by antiadominguez** [reply]
I have the same error changing the host ip

**Reply by ngjaying** [reply]
The error just saying eKuiper cannot access that endpoint. Is the port exported in EdgeX side?

**Reply by antiadominguez** [reply]
I'm running both services in the same machine. I have these ports exported:
![Screenshot from 2024-06-18 10-23-07](https://github.com/lf-edge/ekuiper/assets/165683333/425458ab-8b7c-4073-bcc5-ccb0ffb0fa37)

**Reply by ngjaying** [reply]
notification service is only exported to 127.0.0.1, thus other container cannot acess it. Try to export to 0.0.0.0:59860

**Reply by antiadominguez** [reply]
It does the same. I have the same error

**Reply by ngjaying** [reply]
Use the host name that is defined in the docker compose. Here is an example for command service "url": "http://edgex-core-command:59882/api/v2/device/name/Random-Boolean-Device/WriteBoolValue",

**Reply by antiadominguez** [reply]
That was the solution. Now is doing the http request. Thanks
